plugins/views/frontend/public/heatmap.js

```diff
@@ -730,35 +883,24 @@
 
     function sendXmlHttpRequest(params, apiPath, callback) {
         try {
-            Countly._internals.log("Sending XML HTTP request");
+            log(logLevelEnums.INFO, "[sendXmlHttpRequest] Sending XML HTTP request");
             var xhr = window.XMLHttpRequest ? new window.XMLHttpRequest() : window.ActiveXObject ? new ActiveXObject('Microsoft.XMLHTTP') : null;
 
-            var data = Countly._internals.prepareParams(params);
-            var method = "GET";
-            if (data.length >= 2000) {
-                method = "POST";
-            }
-            else if (Countly.force_post) {
-                method = "POST";
-            }
+            var data = prepareParams(params);
 
-            if (method === "GET") {
-                xhr.open('GET', Countly.passed_data.url + apiPath + "?" + data, true);
-            }
-            else {
-                xhr.open('POST', Countly.passed_data.url + apiPath, true);
-                xhr.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
-            }
-            xhr.setRequestHeader("countly-token", Countly._internals.getToken());
+            xhr.open('POST', Countly.passed_data.url + apiPath, true);
+            xhr.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+            xhr.setRequestHeader("countly-token", Countly._internals.getValueFromStorage('cly_token'));
 
             // fallback on error
             xhr.onreadystatechange = function() {
                 if (this.readyState == this.HEADERS_RECEIVED) {
                     try {
-                        Countly._internals.setToken(xhr.getResponseHeader("content-language"));
+                        Countly._internals.removeValueFromStorage('cly_token');
+                        Countly._internals.setValueInStorage('cly_token', xhr.getResponseHeader("content-language"));
                     }
                     catch (ex) {
-                        Countly._internals.log("failed, trying fallback header");
+                        log(logLevelEnums.ERROR, "[sendXmlHttpRequest] Failed, trying fallback header");
```

Comment:
This logged message doesn't seem to be correctly describing what is happening here.